design/yugabyte/yugabytedb-rook-design.md

```diff
@@ -0,0 +1,89 @@
+YugaByte DB is a high-performance distributed transactional database. (More information [here](https://docs.yugabyte.com/latest/introduction/)). This issue outlines design and interfaces for a Rook YugaByte operator for running YugaByte on Kubernetes with Rook as the underlying storage engine.
+
+**What is use case behind this feature:**
+
+A cluster of YugaByte DB backed by Rook should be created using the `yugabytedbclusters.yugabytedb.rook.io` custom resource definition. Below is a sample custom resource spec for creating a 3-master, 3 t-servers cluster using the CRD. The sample is followed by an explanation of different configuration options available on the YugaByte DB CRD.
```

Comment:
can multiple yugabyte clusters be created per k8s cluster? per k8s namespace?  any conflicts to worry about there?

Reply:
Thanks for bringing this up - multiple YB instances in the same cluster is possible. We just found a small issue with multiple clusters in the same namespace - which we will address in the actual PR.

Reply:
@vishal-biyani - to keep it easy for the end users, it may be best to create one cluster per namespace. We tend to use the namespace as the "cluster name". Will this model create any issues here?

The reason I say this is because we have a way of deploying across multiple k8s clusters (for multi-region use-cases - which we would like to integrate with crossplane for), and that relies on this mental model.